apps/web/components/booking/BookingListItem.tsx

```diff
@@ -12,15 +14,19 @@ import { Dialog, DialogClose, DialogContent, DialogFooter, DialogHeader } from "
 import { TextArea } from "@calcom/ui/form/fields";
 
 import { HttpError } from "@lib/core/http/error";
-import { inferQueryOutput, trpc } from "@lib/trpc";
+import { inferQueryOutput, trpc, inferQueryInput } from "@lib/trpc";
 
 import { useMeQuery } from "@components/Shell";
 import { RescheduleDialog } from "@components/dialog/RescheduleDialog";
 import TableActions, { ActionType } from "@components/ui/TableActions";
 
+type BookingListingStatus = inferQueryInput<"viewer.bookings">["status"];
+
 type BookingItem = inferQueryOutput<"viewer.bookings">["bookings"][number];
 
-function BookingListItem(booking: BookingItem) {
+type BookingItemProps = BookingItem & { listingStatus: BookingListingStatus; recurringCount?: any };
```

Comment:
As mentioned in the code comment, we need to track down the current listing status to be able to act differently on the `upcoming` list versus the new `recurring` list.